### PR TITLE
Recommend and explain datadog-signing-keys version 1.1.0

### PR DESCRIPTION
**Pull request by bkabrda**
### What does this PR do?
Clarifies the situation around datadog-signing-keys package versions and recommends that customers upgrade to 1.1.0.

### Preview

https://docs-staging.datadoghq.com/slavek.kabrda/update-datadog-signing-keys-version/agent/guide/linux-agent-2022-key-rotation

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 1]
```diff
@@ -100,7 +100,12 @@ $ sudo rpm --import /tmp/DATADOG_RPM_KEY_FD4BF915
...
+A host correctly trusts the new key either one of these conditions is true:
+
+- The file `/usr/share/keyrings/datadog-archive-keyring.gpg` exists and the Datadog source list file (usually `/etc/apt/sources.list.d/datadog.list`) contains the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`.
+- The Datadog source list file doesn't contain the `signed-by` option, but `datadog-signing-keys` version 1.1.0 or later is installed, which results in presence of `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` file.
+
+Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by an [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Please ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using a supported [installation method](#install-methods-that-automatically-trust-the-new-gpg-key).
```
Comment (albertvaka) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:108]:
```suggestion
Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by an [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Please ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using one of the [installation method versions listed above](#install-methods-that-automatically-trust-the-new-gpg-key).
```
Intent of the change: use more specific words than "supported"

**Review by albertvaka — APPROVED** [review]
Left one nit but LGTM

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 2]
```diff
@@ -100,7 +100,12 @@ $ sudo rpm --import /tmp/DATADOG_RPM_KEY_FD4BF915
 {{< tabs >}}
 {{% tab "Debian/Ubuntu" %}}
 
-If the file `/usr/share/keyrings/datadog-archive-keyring.gpg` exists, the new key is trusted and no further action is needed. This file is created either by an [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package).
+A host correctly trusts the new key either one of these conditions is true:
+
+- The file `/usr/share/keyrings/datadog-archive-keyring.gpg` exists and the Datadog source list file (usually `/etc/apt/sources.list.d/datadog.list`) contains the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`.
+- The Datadog source list file doesn't contain the `signed-by` option, but `datadog-signing-keys` version 1.1.0 or later is installed, which results in presence of `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` file.
```
Comment (olivielpeau) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:106]: Not a blocker for this PR, but could we later add a script here that can check if either of these conditions is true, and outputs `New GPG key is trusted` (+ `0` exit code) or `New GPG key is NOT trusted` (+ non-`0` exit code). Unless this seems completely useless based on the customer feedback we've already received.

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 3]
```diff
@@ -100,7 +100,12 @@ $ sudo rpm --import /tmp/DATADOG_RPM_KEY_FD4BF915
...
+
+Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by an [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Please ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using one of the [installation method versions listed above](#install-methods-that-automatically-trust-the-new-gpg-key).
```
Comment (olivielpeau) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:108]: Nit

```suggestion
Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by a supported [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Please ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using one of the [installation method versions listed above](#install-methods-that-automatically-trust-the-new-gpg-key).
```

**Review by olivielpeau — APPROVED** [review]
Left a few nits/follow-ups inline, but no blockers, thanks for the update.

For consistency, in the `Manual update` section, I think we should recommend installing `datadog-signing-keys` 1.1.0+, and then provide the current instructions only for cases where customers can't install the datadog-signing-keys package for some reason. (or is there a reason we shouldn't do that?)

File: content/en/agent/guide/linux-agent-2022-key-rotation.md (hunk 2 above)
Comment (bkabrda) [reply content/en/agent/guide/linux-agent-2022-key-rotation.md:106]: So far it doesn't seem necessary, but if the kind of feedback that we're getting changes, I can easily do that.

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 4]
```diff
@@ -153,6 +158,19 @@ The first two methods do not require verification for Datadog's repo metadata, s
...
+
+* On Ubuntu >= 16 and Debian >= 9, only `/usr/share/keyrings/datadog-archive-keyring.gpg` was created, but `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` was not.
+* If the APT source list file (e.g. `/etc/apt/sources.list.d/datadog.list`) didn't contain the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`, APT would never know about the new key and thus any operations with the Datadog repository would fail after the key rotation.
+
+`datadog-signing-keys` version 1.1.0 addresses this issue by creating `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` when `/etc/apt/sources.list.d/datadog.list` doesn't contain the proper `signed-by` option. This ensures that the above corner case is covered as well.
+
+Users of supported up-to-date [installation methods](#install-methods-that-automatically-trust-the-new-gpg-key) using the default Datadog sources always have the proper `signed-by` option configured, so they are not affected by this issue. We highly recommend all other users to upgrade to `datadog-signing-keys` 1.1.0 to ensure preparedness for the upcoming key rotation.
```
Comment (AriDDOG) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:172]: We highly recommend that all other users

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 5]
```diff
@@ -132,7 +137,7 @@ Since Agent v6.31.0 and v7.31.0, all Datadog DEB packages have a soft dependency
 
 Upon installation, this package:
 
-- Configures APT keys in the `/usr/share/keyrings/datadog-archive-keyring.gpg` keyring and also through `apt-key` when necessary. **This ensures that the upcoming APT repository signing key is trusted.**
+- Configures APT keys in the `/usr/share/keyrings/datadog-archive-keyring.gpg` keyring and also in `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` when necessary. **This ensures that the upcoming APT repository signing key is trusted.** [`datadog-signing-keys` version 1.1.0](#datadog-signing-keys-version-110) is recommended to ensure preparedness for the upcoming key rotation.
```
Comment (AriDDOG) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:140]: I think we should say  - "Using the package [`datadog-signing-keys` version 1.1.0](#datadog-signing-keys-version-110) is recommended to ensure ..."

**Review by AriDDOG — COMMENTED** [review]
I think we have the content we need for customers. I suggested minor phrasing updates.

File: content/en/agent/guide/linux-agent-2022-key-rotation.md (hunk 2 above)
Comment (cswatt) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:106]:
```suggestion
- The Datadog source list file doesn't contain the `signed-by` option, but `datadog-signing-keys` version 1.1.0 or later is installed, which results in the presence of a `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` file.
```

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 6]
```diff
@@ -100,7 +100,12 @@ $ sudo rpm --import /tmp/DATADOG_RPM_KEY_FD4BF915
...
+A host correctly trusts the new key either one of these conditions is true:
+
+- The file `/usr/share/keyrings/datadog-archive-keyring.gpg` exists and the Datadog source list file (usually `/etc/apt/sources.list.d/datadog.list`) contains the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`.
+- The Datadog source list file doesn't contain the `signed-by` option, but `datadog-signing-keys` version 1.1.0 or later is installed, which results in presence of `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` file.
+
+Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by a supported [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Please ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using one of the [installation method versions listed above](#install-methods-that-automatically-trust-the-new-gpg-key).
```
Comment (cswatt) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:108]:
```suggestion
Files `/usr/share/keyrings/datadog-archive-keyring.gpg` and, optionally, `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` are created either by a supported [installation method](#install-methods-that-automatically-trust-the-new-gpg-key) or by installing the [`datadog-signing-keys` package](#the-datadog-signing-keys-package). Ensure that `datadog-signing-keys` [version 1.1.0](#datadog-signing-keys-version-110) or later is installed unless using one of the [installation method versions listed above](#install-methods-that-automatically-trust-the-new-gpg-key).
```

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 7]
```diff
@@ -153,6 +158,19 @@ The first two methods do not require verification for Datadog's repo metadata, s
 
 With the third method, you need to explicitly install the `datadog-signing-keys` package if you are installing the Agent package from `apt.datadoghq.com` through `apt`. Alternatively, use one of the supported [installation methods](#install-methods-that-automatically-trust-the-new-gpg-key).
 
+### datadog-signing-keys version 1.1.0
+
+<div class="alert alert-info"><strong>Note:</strong> This section only applies to DEB Agent package users.</div>
+
+`datadog-signing-keys` versions before 1.1.0 didn't handle the following corner case:
```
Comment (cswatt) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:165]:
```suggestion
`datadog-signing-keys` versions before 1.1.0 do not handle the following corner cases:
```

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 8]
```diff
@@ -153,6 +158,19 @@ The first two methods do not require verification for Datadog's repo metadata, s
...
+
+* On Ubuntu >= 16 and Debian >= 9, only `/usr/share/keyrings/datadog-archive-keyring.gpg` was created, but `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` was not.
+* If the APT source list file (e.g. `/etc/apt/sources.list.d/datadog.list`) didn't contain the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`, APT would never know about the new key and thus any operations with the Datadog repository would fail after the key rotation.
```
Comment (cswatt) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:168]:
```suggestion
* If the APT source list file (e.g. `/etc/apt/sources.list.d/datadog.list`) does not contain the option `[signed-by=/usr/share/keyrings/datadog-archive-keyring.gpg]`, APT never knows about the new key. Any operations with the Datadog repository will fail after the key rotation.
```

File: content/en/agent/guide/linux-agent-2022-key-rotation.md [hunk 9]
```diff
@@ -153,6 +158,19 @@ The first two methods do not require verification for Datadog's repo metadata, s
...
+
+`datadog-signing-keys` version 1.1.0 addresses this issue by creating `/etc/apt/trusted.gpg.d/datadog-archive-keyring.gpg` when `/etc/apt/sources.list.d/datadog.list` doesn't contain the proper `signed-by` option. This ensures that the above corner case is covered as well.
+
+Users of supported up-to-date [installation methods](#install-methods-that-automatically-trust-the-new-gpg-key) using the default Datadog sources always have the proper `signed-by` option configured, so they are not affected by this issue. We highly recommend that all other users to upgrade to `datadog-signing-keys` 1.1.0 to ensure preparedness for the upcoming key rotation.
```
Comment (cswatt) [inline content/en/agent/guide/linux-agent-2022-key-rotation.md:172]:
```suggestion
Users of supported up-to-date [installation methods](#install-methods-that-automatically-trust-the-new-gpg-key) using the default Datadog sources always have the proper `signed-by` option configured, so they are not affected by this issue. Datadog highly recommends that all other users to upgrade to `datadog-signing-keys` 1.1.0 to ensure preparedness for the upcoming key rotation.
```

**Review by cswatt — CHANGES_REQUESTED** [review]

**Review by cswatt — APPROVED** [review]

**Pull request merged**